install: Add `--encryption=[ipsec|wireguard]` flag

gandro opened the pull request:
This modifies the `--encryption` flag to a string such that the user can pick a encryption mode, i.e. allowing the user to choose between IPsec or Wireguard.

Enabling Wireguard is basically equivalent to `cilium install --config enable-wireguard=true --config enable-l7-proxy=false`.

Open questions:

- Wireguard in Cilium currently does not work with the L7 proxy. We automatically disable the L7 proxy here if Wireguard is enabled. This might not be needed in future versions of Cilium, I'm not sure what the best approach here would be.
- Old versions of Cilium (e.g. 1.9) do not have support for Wireguard. This means that because the agent is unaware of the `enable-wireguard` ConfigMap option that no encryption will be enabled. Again, I'm not sure what the best approach here should be.

```console
$ cilium install --encryption=wireguard --agent-image=quay.io/cilium/cilium-ci --operator-image=quay.io/cilium/operator-generic-ci --version latest
🔮 Auto-detected Kubernetes kind: kind
✨ Running "kind" validation checks
✅ Detected kind version "0.10.0"
🔮 Auto-detected cluster name: kind-kind
🔮 Auto-detected IPAM mode: kubernetes
ℹ️  kube-proxy-replacement disabled
🔮 Auto-detected datapath mode: tunnel
🔑 Generating CA...
2021/04/20 17:43:25 [INFO] generate received request
2021/04/20 17:43:25 [INFO] received CSR
2021/04/20 17:43:25 [INFO] generating key: ecdsa-256
2021/04/20 17:43:25 [INFO] encoded CSR
2021/04/20 17:43:25 [INFO] signed certificate with serial number 206955789310411371558035536091064166288703354340
🔑 Generating certificates for Hubble...
2021/04/20 17:43:25 [INFO] generate received request
2021/04/20 17:43:25 [INFO] received CSR
2021/04/20 17:43:25 [INFO] generating key: ecdsa-256
2021/04/20 17:43:25 [INFO] encoded CSR
2021/04/20 17:43:25 [INFO] signed certificate with serial number 23733716998220022468916953014748461473981058958
🚀 Creating Service accounts...
🚀 Creating Cluster roles...
🚀 Creating ConfigMap...
ℹ️  L7 proxy disabled
🚀 Creating Agent DaemonSet...
🚀 Creating Operator Deployment...
⌛ Waiting for Cilium to be installed...

```

tgraf commented on internal/cli/cmd/install.go:
```diff
@@ -69,7 +69,8 @@ cilium install --context kind-cluster1 --cluster-id 1 --cluster-name cluster1
 	cmd.Flags().BoolVar(&params.RestartUnmanagedPods, "restart-unmanaged-pods", true, "Restart pods which are not being managed by Cilium")
 	cmd.Flags().BoolVar(&params.Encryption, "encryption", false, "Enable encryption of all workloads traffic")
-	cmd.Flags().BoolVar(&params.NodeEncryption, "node-encryption", false, "Enable encryption of all node to node traffic")
+	cmd.Flags().StringVar(&params.EncryptionMode, "encryption-mode", "ipsec", "Encryption mode { ipsec | wireguard }")
```
I'm also fine to change the `--encryption` flag to be a string. The fewer flags the better. We are still pre 1.0.

tklauser reviewed (approved):
Looks like this needs a rebase after #158 was merged.

The change itself LGTM :rocket:

gandro commented:
> Looks like this needs a rebase after #158 was merged.
> 
> The change itself LGTM rocket

Rebased. And added a warning if the user chooses `--node-encryption` with Wireguard (which we do not support yet in Wireguard, but will in the future)
